eth/downloader/queue.go

```diff
@@ -452,25 +382,19 @@ func (q *queue) countProcessableItems() int {
 		if q.mode == FastSync {
 			bnum := result.Header.Number.Uint64()
 			if bnum == q.fastSyncPivot {
-				// If the state of the pivot block is not
-				// available yet, we cannot proceed and return 0.
-				//
 				// Stop before processing the pivot block to ensure that
 				// resultCache has space for fsHeaderForceVerify items. Not
 				// doing this could leave us unable to download the required
 				// amount of headers.
-				if i > 0 || len(q.stateTaskPool) > 0 || q.pendingNodeDataLocked() > 0 {
-					return i
-				}
```

Comment:
Again a tiny invariant change. I don't yet see how this could have an adverse effect, just thought I'd mention it. Previously the code handled the pivot block fully independently from any other blocks. The new code places the pivot block possibly to the end of a fast sync import batch. It **should** work, just mentioning the behavioral change.